### PR TITLE
Switch CI to use ros-industrial master

### PR DESCRIPTION
**Pull request by mpowelson**
The fix needed in ipa-mdl's branch has since been merged.

### CODE REVIEW

**Comment by mpowelson** [comment]
@Levi-Armstrong Well using master does not work the same way ipl-mds's branch used to. Trying to decipher the comments it looks like I need to do something to the docker. In the meantime, I switched it to the target workspace instead of upstream. Hopefully that will fix it.

**Comment by mpowelson** [comment]
Well tesseract_python is broken. I don't think I am going to spend time fixing this right now.

**Comment by ipa-mdl** [comment]
@mpowelson: Your Docker image contains outdated versions of colcon/ament/rosdep (not sure, which one was fixed).

If cannot fix the image, you might try an upgrade: `AFTER_INIT: apt-get upgrade -y`.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
              OS_CODE_NAME: bionic,
              ROS_DISTRO: eloquent,
              ROS_REPO: main,
-             UPSTREAM_WORKSPACE: 'dependencies.rosinstall',
+             TARGET_WORKSPACE: 'dependencies.rosinstall',
```
Comment (ipa-mdl) [inline .github/workflows/main.yml:28]: This omits your target repository, `UPSTREAM_WORKSPACE` should work fine with a newer build system.

**Comment by ipa-mdl** [comment]
Something is wrong with the upstream_ws or how it is built. `ros2 pkg list` does not list any of these packages.

**Comment by ipa-mdl** [comment]
Are all tesseract packages built with `cmake` only? I think they are not added to the ROS2 index.

**Comment by mpowelson** [comment]
Yes. They are all ROS agnostic and do not use Ament, so I think that is correct. I don't think they are added to the index unless the ament_index file is manually created like is being done in [tesseract_collision](https://github.com/ros-industrial-consortium/tesseract/blob/b508c891d063934d2e5028fc40986c18c274b437/tesseract/tesseract_collision/CMakeLists.txt#L130)

**Comment by ipa-mdl** [comment]
I guess you build is failing now because of https://github.com/ros-infrastructure/rosdep/issues/724

**Comment by Levi-Armstrong** [comment]
@schornakj Didn't we add the creation of these for ROS2 support to Tesseract?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -28,14 +29,15 @@ jobs:
              UPSTREAM_WORKSPACE: 'dependencies.rosinstall',
```
Comment (ipa-mdl) [inline .github/workflows/main.yml:29]:
```suggestion
             TARGET_WORKSPACE: '. dependencies.rosinstall',
```

The dot means target repository.

Comment (ipa-mdl) [reply .github/workflows/main.yml:29]: this puts all packages into the targer workspace, but it will build and run all tests as well.

**Comment by Levi-Armstrong** [comment]
It looks like we only do this for tesseract_collision. We should add this to the `tesseract_configure_package` so all get added.

**Comment by Levi-Armstrong** [comment]
This is what we are doing for pure cmake packages.

```
# Create an ament_index resource file to allow ROS2 ament_index_cpp to locate the installed path to this package.
# This is a workaround to let the ROS2 version of pluginlib find tesseract_collision's plugins, since tesseract_collision is a non-ROS CMake package.
# ADDITIONAL REQUIREMENT: The installed path must be added to the AMENT_PREFIX_PATH environment variable at runtime, which is outside the scope of CMakeLists.txt.
file(WRITE ${CMAKE_INSTALL_PREFIX}/share/ament_index/resource_index/packages/${PROJECT_NAME} "")
```

**Comment by ipa-mdl** [comment]
> ADDITIONAL REQUIREMENT: The installed path must be added to the AMENT_PREFIX_PATH environment variable at runtime, which is outside the scope of CMakeLists.txt

That's needs to be done somehow
I could come up with a solution like https://github.com/ros-industrial/industrial_ci/pull/496, but it would still be broken for others..

**Comment by ipa-mdl** [comment]
I am currently working on a generic fix (https://github.com/ros-industrial/industrial_ci/pull/496).

**Comment by mpowelson** [comment]
Thanks. It seems like ultimately this needs to be fixed in rosdep.

**Comment by schornakj** [comment]
> @schornakj Didn't we add the creation of these for ROS2 support to Tesseract?

That was originally a solution to the way ROS2 `pluginlib` finds plugins, which relies on `ament_index_cpp` to find packages using the ament package index. It was only needed for the `tesseract_collision` package, since that was the only Tesseract package that contained plugins.

I agree that it's a shortcoming of `rosdep`, as well as `pluginlib`.

**Comment by ipa-mdl** [comment]
> I am currently working on a generic fix (ros-industrial/industrial_ci#496).

I have merged the fix into the `master` branch.
It is not ideal, but better than a lot of unexpected CI failures.

80affcc29d72d2015c1a72a6a0cfbf5439fc37cd should work now without the `AFTER_INIT` script

**Comment by Levi-Armstrong** [comment]
@schornakj It looks like there may be solution in Eloquent explained [here](https://github.com/ros-infrastructure/rosdep/issues/724). It looks like we just need to install two new files along with the index file for each package. Would you have time to test this and see if it eliminates the need to modify the AMENT_PREFIX_PATH? @ipa-mdl This may also solve the rosdep issue for pure cmake packages. We will give it a try.

**Comment by gavanderhoorn** [comment]
@schornakj wrote:
> That was originally a solution to the way ROS2 `pluginlib` finds plugins, which relies on `ament_index_cpp` to find packages using the ament package index. It was only needed for the `tesseract_collision` package, since that was the only Tesseract package that contained plugins.
> 
> I agree that it's a shortcoming of `rosdep`, as well as `pluginlib`.

off-topic, but: if a plain CMake pkg wants to contribute dynamic libraries which are to be considered ROS plugins for other ROS packages, it makes sense they need to be made visible to the ROS infrastructure for loading those plugins.

To avoid excessive file system crawling, the index was created.

So for non-ROS pkgs to provide plugins, the loadable artefacts must be registered in the index.

That doesn't seem like a shortcoming necessarily, just something which is different from ROS 1, combined with the fact you're now mostly using non-ROS pkgs.

**Comment by schornakj** [comment]
@Levi-Armstrong @mpowelson @gavanderhoorn Does anything else need to happen with this PR before we can merge it?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -16,26 +16,28 @@ jobs:
              OS_CODE_NAME: bionic,
              ROS_DISTRO: eloquent,
              ROS_REPO: main,
-             ROSDEP_SKIP_KEYS: "bullet3 fcl ompl",
+             ROSDEP_SKIP_KEYS: "bullet3 fcl ompl tesseract tesseract_collision tesseract_common tesseract_environment tesseract_geometry tesseract_kinematics tesseract_motion_planners tesseract_process_planners tesseract_scene_graph tesseract_support tesseract_urdf tesseract_visualization",
```
Comment (Levi-Armstrong) [inline .github/workflows/main.yml:19]: Is this required now with the fixes to tesseract updating the prefix path?

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -16,26 +16,28 @@ jobs:
...
+             AFTER_INIT: "apt-get upgrade -y",
              BADGE: clang-format}
           - {OS_NAME: ubuntu,
              OS_CODE_NAME: bionic,
              ROS_DISTRO: eloquent,
              ROS_REPO: main,
              UPSTREAM_WORKSPACE: 'dependencies.rosinstall',
-             ROSDEP_SKIP_KEYS: "bullet3 fcl ompl",
+             ROSDEP_SKIP_KEYS: "bullet3 fcl ompl tesseract tesseract_collision tesseract_common tesseract_environment tesseract_geometry tesseract_kinematics tesseract_motion_planners tesseract_process_planners tesseract_scene_graph tesseract_support tesseract_urdf tesseract_visualization",
              DOCKER_IMAGE: "mpowelson/tesseract:eloquent",
```
Comment (Levi-Armstrong) [inline .github/workflows/main.yml:31]: same as above

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -47,13 +49,14 @@ jobs:
...
```
Comment (Levi-Armstrong) [inline .github/workflows/main.yml:53]: same as above

**Comment by mpowelson** [comment]
@Levi-Armstrong Could we get the [foxy build](https://github.com/ros-industrial/docker/pull/35) added to the dockerhub, so I could modify this to use that instead?

File: .github/workflows/main.yml (hunk 3 above)
Comment (mpowelson) [reply .github/workflows/main.yml]: No. It worked without it on [mine](https://github.com/mpowelson/tesseract_ros2/actions/runs/165046421). I just removed it and added cmake_common_scripts

**Comment by Minipada** [comment]
I have the changes ready, will push as soon as it's on dockerhub

**Comment by Levi-Armstrong** [comment]
I have fixed the docker hub hooks in the docker repository for tesseract. Foxy is currently building but the other three have successfully built.

**Comment by Levi-Armstrong** [comment]
Foxy Docker is now available.

**Comment by mpowelson** [comment]
I will not be able to mess with this more any time soon to add foxy, but could we merge this so we stop getting nightly emails that ci is broken?

**Comment by mpowelson** [comment]
Thanks

**Pull request merged**